python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -470,6 +584,35 @@ def _calculate_used_data(
 
         return known_data
 
+    @cached_method
+    def _calculate_used_data(
+        self,
+        asset_graph: AssetGraph,
+        asset_key: AssetKey,
+        record_id: Optional[int],
+        record_timestamp: Optional[float],
+        record_tags: Mapping[str, str],
+    ) -> Dict[AssetKey, Tuple[Optional[int], Optional[float]]]:
+        if record_id is None:
+            return {key: (None, None) for key in asset_graph.get_non_source_roots(asset_key)}
+
+        partitions_def = asset_graph.get_partitions_def(asset_key)
+        if isinstance(partitions_def, TimeWindowPartitionsDefinition):
+            return self._calculate_used_data_partitioned(
+                asset_graph=asset_graph,
+                asset_key=asset_key,
+                record_id=record_id,
+                partitions_def=partitions_def,
+            )
+        else:
```

Comment:
Should we explicitly handle the case where it's partitioned, but not time-partitioned? Raise an error?

Reply:
Definitely reasonable, I'm inclined to leave it as is for now to avoid changing existing behavior.

Reply:
Can we address this in a followup then?

Reply:
Or I guess - what is the existing behavior?

Reply:
What is throwing me off here is that we're calling `_calculate_used_data_unpartitioned` on an asset that might be partitioned

Reply:
yeah definitely can address in a followup -- existing behavior was that we would treat a static-partitioned asset as an unpartitioned asset. This is not correct behavior, but throwing an exception may break existing workflows that "kinda work". Would rather have a smoother landing here than break it